marlowe-playground-client/src/Blockly/Dom.purs

```diff
@@ -0,0 +1,253 @@
+--                             <block type="TrueObservationType" id="@}aK]jjhxi=2LLj*5g}."></block>
+--                         </value>
+--                         <statement name="contract">
+--                             <block type="CloseContractType" id="Yd_ab!Vtyb88H@?Eqj9E"></block>
+--                         </statement>
+--                         <next>
+--                             <block type="ChoiceActionType" id=")z[fYD/@_GoEtKS}/48t">
+--                                 <field name="choice_name">name</field>
+--                                 <value name="party">
+--                                     <block type="RolePartyType" id="`blYZ~JOFle+tRT_@;KS">
+--                                         <field name="role">role</field>
+--                                     </block>
+--                                 </value>
+--                                 <statement name="bounds">
+--                                     <block type="BoundsType" id="5x$vp,OGm^;CWEYL;l_u">
+--                                         <field name="from">0</field>
+--                                         <field name="to">0</field>
+--                                     </block>
+--                                 </statement>
+--                                 <statement name="contract">
+--                                     <block type="CloseContractType" id="VkYy9=B7eZ/AZAd@6jr)"></block>
+--                                 </statement>
+--                             </block>
+--                         </next>
+--                     </block>
+--                 </statement>
+--                 <statement name="contract">
+--                     <block type="CloseContractType" id="dzm~uz;1}9ZF,+L^UbCX"></block>
+--                 </statement>
+--             </block>
+--         </statement>
+--     </block>
+-- </xml>
+module Blockly.Dom where
+
+import Prelude
+import Blockly.Internal (workspaceToDom)
+import Blockly.Types (BlocklyState)
+import Control.Monad.Error.Extra (toMonadThrow)
+import Control.Monad.Except (runExceptT, throwError)
+import Control.Monad.Except.Trans (class MonadThrow)
+import Data.Array (find, length)
+import Data.Array.Partial as UnsafeArray
+import Data.Compactable (separate)
+import Data.Either (Either(..), note')
+import Data.Lens (Lens', _1, view)
+import Data.Lens.Iso.Newtype (_Newtype)
+import Data.Lens.Record (prop)
+import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Data.Symbol (SProxy(..))
+import Data.Traversable (traverse)
+import Data.Tuple (Tuple(..))
+import Effect (Effect)
+import Effect.Class (class MonadEffect, liftEffect)
+import Foreign.Object (Object)
+import Foreign.Object as Object
+import Partial.Unsafe (unsafePartial)
+import Web.DOM (Element, Node)
+import Web.DOM.Element as Element
+import Web.DOM.HTMLCollection as HTMLCollection
+import Web.DOM.Node as Node
+import Web.DOM.NodeList as NodeList
+import Web.DOM.ParentNode as ParentNode
+
+newtype Block
```

Comment:
does this need to be a `newtype` and not just a `type`?

Reply:
yeah, it could be a `type`, not sure if it hurts though, and I think it makes stuff like this more readable

```purescript
        $ Tuple
            (Block { id: blockId, type: blockType, children })
            (join right)
```

Should I change it?

Reply:
I just find that in Purescript (not Haskell) we should only use a newtype around a record if there is some typeclass that we need to define for it. I would prefer it to be changed but it's up to you if you think it helps readability.